sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala

```diff
@@ -327,7 +327,127 @@ class HiveQuerySuite extends HiveComparisonTest with BeforeAndAfter {
 
   createQueryTest("transform",
     "SELECT TRANSFORM (key) USING 'cat' AS (tKey) FROM src")
+ 
+  test("schema-less transform") {
```

Comment:
As we are targeting the compatibility of Hive, let's make the test case via `createQueryTest`, which probably more clear.